hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconStorageContainerManagerFacade.java

```diff
@@ -400,18 +451,200 @@ private void initializeSCMDB() {
   }
 
   public void updateReconSCMDBWithNewSnapshot() throws IOException {
-    DBCheckpoint dbSnapshot = scmServiceProvider.getSCMDBSnapshot();
-    if (dbSnapshot != null && dbSnapshot.getCheckpointLocation() != null) {
-      LOG.info("Got new checkpoint from SCM : " +
-          dbSnapshot.getCheckpointLocation());
+    if (isSyncDataFromSCMRunning.compareAndSet(false, true)) {
+      DBCheckpoint dbSnapshot = scmServiceProvider.getSCMDBSnapshot();
+      if (dbSnapshot != null && dbSnapshot.getCheckpointLocation() != null) {
+        LOG.info("Got new checkpoint from SCM : " +
+            dbSnapshot.getCheckpointLocation());
+        try {
+          initializeNewRdbStore(dbSnapshot.getCheckpointLocation().toFile());
+        } catch (IOException e) {
+          LOG.error("Unable to refresh Recon SCM DB Snapshot. ", e);
+        }
+      } else {
+        LOG.error("Null snapshot location got from SCM.");
+      }
+    } else {
+      LOG.warn("SCM DB sync is already running.");
+    }
+  }
+
+  public boolean syncSCMContainerInfoWithReconContainerInfo()
+      throws IOException {
+    if (isSyncDataFromSCMRunning.compareAndSet(false, true)) {
       try {
-        initializeNewRdbStore(dbSnapshot.getCheckpointLocation().toFile());
+        List<ContainerInfo> containers = containerManager.getContainers();
+        List<ContainerInfo> listOfContainers = scmServiceProvider.
+            getListOfContainers();
+        if (null != listOfContainers && listOfContainers.size() > 0) {
+          LOG.info("Got list of containers frm SCM : " +
+              listOfContainers.size());
+          listOfContainers.forEach(containerInfo -> {
+            try {
+              long containerID = containerInfo.getContainerID();
+              List<HddsProtos.SCMContainerReplicaProto> containerReplicas
+                  = scmServiceProvider.getContainerReplicas(containerID);
```

Comment:
This will make communication stuck for long time, if there are 10,000 of containers or much higher, including closed containers.
We need a kind of FCR report from SCM in a iterative manner to support this with container information.

Reply:
@swagle @kerneltime , pls provide your comments on above solution suggested by @sumitagrawl